htmlTrim.js

```diff
@@ -0,0 +1,48 @@
+import textToHtml from './textToHtml';
+
+/**
+ * Remove empty nodes from given html as an argument. 
+ * @param {*} html The template string OR The template node
+ * @param {Boolean} deep Deeply remove the empty child or not.
+ */
+const removeEmptyNodes = (html, deep) => {
+  if (typeof html === 'string') {
+    html = textToHtml(html);
+  }
+
+  let children = Array.from(html.children || []);
+  children.forEach(el => {
+    if (el.textContent.trim() === "") {
+      html.removeChild(el)
+    } else {
+      if (deep && el && el.children && el.children.length > 0) {
+        removeEmptyNodes(el, deep);
+      }
+    }
+  });
+
+  return html;
+}
+
+/**
+ * Remove empty nodes from given html as an argument. 
+ * @param {*} html The template string OR The template node
+ * @param {Boolean} deep Deeply remove the empty child or not.
+ */
+export const htmlTrim = (html, deep) => {
```

Comment:
The way library is written doesn't match our standards. 
- It's documentation is missing
- For any public function/module, first try to define it's behaviors. And Write a clean logic which satisfies all of the behavioral criteria. 

In the documentation we will describe the function's behavior based on the defined behavioral criteria. We had a verbal discussion on it when it was planned. But, later you missed it.